lib/Zonemaster/CLI.pm

```diff
@@ -839,16 +851,49 @@ sub add_fake_delegation {
 
         if ( $ip ) {
             my $net_ip = Net::IP::XS->new( $ip );
-            if ( ( $ip =~ /($IPV4_RE)/ && Net::IP::XS::ip_is_ipv4( $ip ) )
-                or
-                 ( $ip =~ /($IPV6_RE)/ && Net::IP::XS::ip_is_ipv6( $ip ) )
-            ) {
-                push @{ $data{ $name } }, $ip;
-            }
-            else {
+            if ( not(
+                     ( $ip =~ /($IPV4_RE)/ && Net::IP::XS::ip_is_ipv4( $ip ) )
+                      or
+                     ( $ip =~ /($IPV6_RE)/ && Net::IP::XS::ip_is_ipv6( $ip ) )
+                    )
+            )
+	    {
                 die Net::IP::XS::Error() ? "Invalid IP address in --ns argument:\n\t". Net::IP::XS::Error() ."\n" : "Invalid IP address in --ns argument.\n";
             }
         }
+    }
+
+    return;
+
+}
+
+sub check_fake_ds {
+    my ( $self, $domain ) = @_;
+    my @data;
+
+    foreach my $str ( @{ $self->ds } ) {
+
+        if ( not $str =~ /$DS_RE/g ) {
```

Comment:
Is the `g` flag really necessary for the regular expression?

Reply:
no, fixed and i am using `unless` instead of `if ( not(`.